project/context.go

```diff
@@ -36,32 +36,35 @@ type Context struct {
 	Project             *Project
 }
 
-func (c *Context) readComposeFile() error {
+func (c *Context) readComposeFiles() error {
```

Comment:
This method feels a little bit complicated to me (too much complexity / indentation).

```go
func (c *Context) readComposeFiles() error {
	if c.ComposeBytes != nil {
		return nil
	}

	logrus.Debugf("Opening compose files: %s", strings.Join(c.ComposeFiles, ","))

	// Handle STDIN (`-f -`)
	if len(c.ComposeFiles) == 1 && c.ComposeFiles[0] == "-" {
		composeBytes, err := ioutil.ReadAll(os.Stdin)
		if err != nil {
			logrus.Errorf("Failed to read compose file from stdin: %v", err)
			return err
		}
		c.ComposeBytes = [][]byte{composeBytes}
		return nil
	}

	for _, composeFile := range c.ComposeFiles {
		composeBytes, err := ioutil.ReadFile(composeFile)
		if err != nil {
			if os.IsNotExist(err) && !c.IgnoreMissingConfig {
				logrus.Errorf("Failed to find %s", composeFile)
				return err
			}
			logrus.Errorf("Failed to open %s", composeFile)
			return err
		}
		c.ComposeBytes = append(c.ComposeBytes, composeBytes)
	}

	return nil
}
```

or even :

```go
func (c *Context) readComposeFiles() error {
	if c.ComposeBytes != nil {
		return nil
	}

	logrus.Debugf("Opening compose files: %s", strings.Join(c.ComposeFiles, ","))

	// Handle STDIN (`-f -`)
	if len(c.ComposeFiles) == 1 && c.ComposeFiles[0] == "-" {
		composeBytes, err := readComposeFile(os.Stdin)
		if err != nil {
			return err
		}
		c.ComposeBytes = [][]byte{composeBytes}
		return nil
	}

	for _, composeFile := range c.ComposeFiles {
		composeFileF, err := os.Open(composeFile)
		if os.IsNotExist(err) && !c.IgnoreMissingConfig {
			logrus.Errorf("Failed to find %s", composeFile)
			return err
		}
		composeBytes, err := readComposeFile(composeFileF)
		if err != nil {
			return err
		}
		c.ComposeBytes = append(c.ComposeBytes, composeBytes)
	}

	return nil
}

func readComposeFile(reader io.Reader) ([]byte, error) {
	composeBytes, err := ioutil.ReadAll(reader)
	if err != nil {
		logrus.Errorf("Failed to read compose file : %v", err)
		return nil, err
	}
	return composeBytes, nil
}
```

After looking at both, I prefer the first one :smiley_cat:.